test/jdk/sun/security/provider/acvp/ML_DSA_Test.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import jdk.test.lib.Asserts;
+import jdk.test.lib.json.JSONValue;
+import jdk.test.lib.security.FixedSecureRandom;
+
+import java.security.*;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.NamedParameterSpec;
+
+import static jdk.test.lib.Utils.toByteArray;
+
+// JSON spec at https://pages.nist.gov/ACVP/draft-celi-acvp-ml-dsa.html
+public class ML_DSA_Test {
+
+    public static void run(JSONValue kat, Provider provider) throws Exception {
+        var mode = kat.get("mode").asString();
+        switch (mode) {
+            case "keyGen" -> mldsaGen(kat, provider);
+            case "sigGen" -> mldsaSign(kat, provider);
+            case "sigVer" -> mldsaVerify(kat, provider);
+            default -> throw new UnsupportedOperationException("Unknown mode: " + mode);
+        }
+    }
+
+    static void mldsaGen(JSONValue kat, Provider p) throws Exception {
```

Comment:
Since this class is called MD_DSA_Test, is it redundant to use "mldsa" in every method name? It might be cleaner to just call then "generate", "sign", and "verify".

Reply:
Sure. I'll modify the names. These were originally one big file.